tests/FsUnit.Xunit.Test/typed.shouldEqualTests.fs

```
@@ -0,0 +1,130 @@
+
+    [<Fact>]
+    member __.``reference type should fail to equal other``() =
+        shouldFail(fun () -> anObj |> shouldEqual otherObj)
+
+    [<Fact>]
+    member __.``reference type should not equal other``() =
+        anObj |> shouldNotEqual otherObj
+
+    [<Fact>]
+    member __.``reference type should fail to not equal itself``() =
+        shouldFail(fun () -> anObj |> shouldNotEqual anObj)
+
+    [<Fact>]
+    member __.``should pass when Equals returns true``() =
+        anObj |> shouldEqual(box(new AlwaysEqual()))
+
+    [<Fact>]
+    member __.``should fail when Equals returns false``() =
+        shouldFail(fun () -> anObj |> shouldEqual(box(new NeverEqual())))
+
+    [<Fact>]
+    member __.``should pass when negated and Equals returns false``() =
+        anObj |> shouldNotEqual(box(new NeverEqual()))
+
+    [<Fact>]
+    member __.``should fail when negated and Equals returns true``() =
+        shouldFail(fun () -> anObj |> shouldNotEqual(box(new AlwaysEqual())))
+
+    [<Fact>]
+    member __.``None should equal None``() =
+        None |> shouldEqual None
+
+    [<Fact>]
+    member __.``Error "Foo" should equal Error "Foo"``() =
+        Error "Foo" |> shouldEqual(Error "Foo")
+
+    [<Fact>]
+    member __.``Error "Foo" should equal fails and have same message``() =
+        (fun () -> Error "Foo" |> shouldEqual(Error "Bar"))
+        |> Assert.Throws<System.NullReferenceException>
+        |> fun e ->
+            e.Message
+            |> shouldEqual(sprintf "  Expected: Error \"Bar\" or Error \"Bar\"%s  But was:  Error \"Foo\"%s" Environment.NewLine Environment.NewLine)
+
+    [<Fact>]
+    member __.``Error "Foo" should not equal Error "Bar"``() =
+        Error "Foo" |> shouldNotEqual(Error "Bar")
+
+    [<Fact>]
+    member __.``Error "Foo" should not equal Error "Bar" fails and have same message``() =
+        (fun () -> Error "Foo" |> shouldNotEqual(Error "Foo"))
+        |> Assert.Throws<System.NullReferenceException>
```

`MatchException` is thrown instead of `NullReferenceException `